File: src/main/java/jugglinglab/jml/JMLAttributes.java
//
// JMLAttributes.java
//
// Copyright 2002-2025 Jack Boyce and the Juggling Lab contributors
//

package jugglinglab.jml;

import java.util.*;

public class JMLAttributes {
  protected JMLNode parentTag;
  protected int size;
  protected ArrayList<String> names;
  protected ArrayList<String> values;

  public JMLAttributes(JMLNode parent) {
    parentTag = parent;
    size = 0;
  }

  public int getNumberOfAttributes() {
    return size;
  }

  public void addAttribute(String name, String value) {
    if (size == 0) {
      names = new ArrayList<>();
      values = new ArrayList<>();
    }
    names.add(name);
    values.add(value);
    size++;
  }

  public String getAttributeName(int index) {
    return names.get(index);
  }

  public String getAttributeValue(int index) {
    return values.get(index);
  }

  public String getAttribute(String name) {
    for (int i = 0; i < size; i++)
      if (name.equalsIgnoreCase(getAttributeName(i))) {
        return getAttributeValue(i);
      }
    return null;
  }
}

File: src/main/kotlin/jugglinglab/jml/JMLAttributes.kt
//
// JMLAttributes.java
//
// Copyright 2002-2025 Jack Boyce and the Juggling Lab contributors
//

package jugglinglab.jml

class JMLAttributes(private var parentTag: JMLNode?) {
    var numberOfAttributes: Int = 0
        private set
    private var names: MutableList<String> = ArrayList()
    private var values: MutableList<String> = ArrayList()

    fun addAttribute(name: String, value: String) {
        names.add(name)
        values.add(value)
        numberOfAttributes++
    }

    fun getAttributeName(index: Int) = names[index]

    fun getAttributeValue(index: Int) = values[index]

    fun getAttribute(name: String): String? {
        for (i in 0..<this.numberOfAttributes) {
            if (name.equals(names[i], ignoreCase = true)) {
                return values[i]
            }
        }
        return null
    }
}
